include/npcomp/Dialect/Basicpy/IR/BasicpyOps.td

```diff
@@ -90,53 +91,53 @@ def CompareOperationAttr : StrEnumAttr<
 }
 
 //===----------------------------------------------------------------------===//
-// Operations
+// Constant and constructor operations
 //===----------------------------------------------------------------------===//
 
-def Basicpy_BinaryCompareOp : Basicpy_Op<"binary_compare", []> {
-  let summary = "Performs a comparison between two operands";
+def Basicpy_NumericConstantOp : Basicpy_Op<"numeric_constant", [
+    ConstantLike, NoSideEffect, DeclareOpInterfaceMethods<OpAsmOpInterface>]> {
+  let summary = "A constant from the Python3 numeric type hierarchy";
   let description = [{
-    This op performs only one step of a potentially multi-step short
-    circuit comparison.
-    See: https://docs.python.org/3/reference/expressions.html#comparisons
+    Basicpy re-uses core MLIR types to represent the Python3 numeric type
+    hierarchy with the following mappings:
+
+    * Python3 `int` : In python, this type is signed, arbitrary precision but
+      in typical realizations, it maps to an MLIR `IntegerType` of a fixed
+      bit-width (typically si64 if no further information is known). In the
+      future, there may be a real `Basicpy::IntType` that retains the true
+      arbitrary precision nature, but this is deemed an enhancement that
+      does not obviate the need to infer physical, sized types for many
+      real-world cases. As such, the Basicpy numeric type hierarchy will
+      always include physical `IntegerType`, if only to enable progressive
+      lowering and interop with cases where the precise type is known.
+    * Python3 `float` : This is allowed to map to any legal floating point
+      type on the physical machine and is usually represented as a double (f64).
+      In MLIR, any `FloatType` is allowed, which facilitates progressive
+      lowering and interop with cases where a more precise type is known.
+    * Python3 `complex` : Maps to an MLIR `ComplexType` with a `FloatType`
+      elementType (note: in Python, complex numbers are always defined with
+      floating point components).
+    * `bool` : See `bool_constant`. In Python, bool is part of the numeric
```

Comment:
what is the relationship to bool_constant? Does this op supercede it?

Reply:
Commented.